### PR TITLE
Suuchi partitioner

### PR DESCRIPTION
**Pull request by brewkode**
Has the following 
- basic partitioner sketch
- Consistent Hash Ring implementation
- CHR Spec
- Partitioner Implementation using CHR - which doesn't take into account replication.

### CODE REVIEW

**Comment by brewkode** [comment]
Notes:
- CHR can be made immutable
- CHR find isn't tested. Needs support from Hash. Will fix it in sometime.

**Comment by brewkode** [comment]
Fixes #9

File: src/test/scala/in/ashwanthkumar/suuchi/partitioner/ConsistentHashRingSpec.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+object IdentityHash extends Hash {
+  override def hash(bytes: Array[Byte]): Integer = bytes.toString.toInt
+}
+
+class ConsistentHashRingSpec extends FlatSpec {
+  "ConsistentHashRing" should "pin nodes into the ring accounting for virtual nodes" in {
+    val ring = new ConsistentHashRing(SuuchiHash, 3)
```
Comment (ashwanthkumar) [inline src/test/scala/in/ashwanthkumar/suuchi/partitioner/ConsistentHashRingSpec.scala:12]: @brewkode Did you intend to use `IdentityHash` here?

File: src/main/scala/in/ashwanthkumar/suuchi/partitioner/PartitionNode.scala [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package in.ashwanthkumar.suuchi.partitioner
+
+case class Node(id: Int, host: String)
```
Comment (ashwanthkumar) [inline src/main/scala/in/ashwanthkumar/suuchi/partitioner/PartitionNode.scala:3]: Can we also have `port: Int` here so it's useful while forwarding?

Comment (ashwanthkumar) [reply src/main/scala/in/ashwanthkumar/suuchi/partitioner/PartitionNode.scala:3]: Thinking more - should we reuse the existing `MemberAddress` for `Node` ?  or re-use `Member` and `MemberAddress` inside `Node` ?

File: src/test/scala/in/ashwanthkumar/suuchi/partitioner/ConsistentHashRingSpec.scala (hunk 1 above)
Comment (brewkode) [reply src/test/scala/in/ashwanthkumar/suuchi/partitioner/ConsistentHashRingSpec.scala:12]: @ashwanthkumar Actually, No. I might have to change things a little bit there to get better tests going. Will chip away at it in parallel

File: src/main/scala/in/ashwanthkumar/suuchi/partitioner/PartitionNode.scala (hunk 2 above)
Comment (brewkode) [reply src/main/scala/in/ashwanthkumar/suuchi/partitioner/PartitionNode.scala:3]: I thought about it too. If at all anything, We should use be using `MemberAddress`. But, I will do it once I merge to master. ok?

**Comment by brewkode** [comment]
Merging this @ashwanthkumar.. Lets do another pass at refactoring some of these in the upcoming wave...

**Pull request merged**